apps/patient/src/views/Pharmacy.tsx

```diff
@@ -47,56 +48,97 @@ const PHARMACY_SEARCH_RADIUS_IN_MILES = 25;
 
-  const orgSettings = getSettings(order.organization.id);
+  const orgSettings = getSettings(order?.organization.id);
 
   const navigate = useNavigate();
+  const toast = useToast();
 
+  // search params
   const [searchParams] = useSearchParams();
   const token = searchParams.get('token');
   const isReroute = searchParams.get('reroute');
   const openNow = searchParams.get('openNow');
   const isDemo = searchParams.get('demo');
   const phone = searchParams.get('phone');
+
+  // preferred pharmacy
   const [preferredPharmacyId, setPreferredPharmacyId] = useState<string>('');
   const [savingPreferred, setSavingPreferred] = useState<boolean>(false);
-  const [pharmacyOptions, setPharmacyOptions] = useState([]);
+
+  // View state
   const [showFooter, setShowFooter] = useState<boolean>(false);
-  const [selectedId, setSelectedId] = useState<string>('');
   const [locationModalOpen, setLocationModalOpen] = useState<boolean>(false);
+
+  // selection state
+  const [selectedId, setSelectedId] = useState<string>('');
+
+  // Submitting state
   const [submitting, setSubmitting] = useState<boolean>(false);
   const [successfullySubmitted, setSuccessfullySubmitted] = useState<boolean>(false);
+
+  // loading state
   const [loadingPharmacies, setLoadingPharmacies] = useState<boolean>(false);
   const [showingAllPharmacies, setShowingAllPharmacies] = useState<boolean>(false);
-  const [latitude, setLatitude] = useState<number | undefined>(undefined);
-  const [longitude, setLongitude] = useState<number | undefined>(undefined);
-  const [location, setLocation] = useState<string>(
-    order?.address ? formatAddress(order.address) : ''
+
+  // Address state
+  const [latitude, setLatitude] = useState<number>();
+  const [longitude, setLongitude] = useState<number>();
+  const [location, setLocation] = useState(
+    order?.address ? formatAddress(order.address) : undefined
   );
+  const [cleanAddress, setCleanAddress] = useState<string>();
+  const [loadingLocation, setLoadingLocation] = useState(false);
+
+  // loading state
+  const isLoading = useMemo(
+    () => loadingLocation || loadingPharmacies,
+    [loadingLocation, loadingPharmacies]
+  );
+
+  // filters
   const [enableOpenNow, setEnableOpenNow] = useState(
     openNow !== null ? !!openNow : order?.readyBy === 'Urgent'
   );
   const [enable24Hr, setEnable24Hr] = useState(order?.readyBy === 'After hours');
 
-  const toast = useToast();
+  // pagination
+  const [pageOffset, setPageOffset] = useState(0);
 
-  const isMultiRx = flattenedFills.length > 1;
+  const isMultiRx = useMemo(() => flattenedFills.length > 1, [flattenedFills.length]);
 
-  const enableMailOrder = !isDemo && orgSettings.mailOrderNavigate;
-  const enableTopRankedCostco = !isDemo && orgSettings.topRankedCostco;
-  const enableTopRankedWalgreens = !isDemo && orgSettings.topRankedWalgreens;
-  const containsGLP = flattenedFills.some((fill) => isGLP(fill.treatment.name));
+  const enableMailOrder = useMemo(
```

Comment:
useMemo here might be overkill https://blog.logrocket.com/when-not-to-use-usememo-react-hook/

"Don’t use useMemo if your operation is inexpensive"

Reply:
Hmmm why did I use it... Probably there was a reason but yeah you're right